streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSessionStore.java

```diff
@@ -0,0 +1,196 @@
+                                        aggSerde == null ? (Serde<AGG>) context.valueSerde() : aggSerde);
+
+        bytesStore.init(context, root);
+    }
+
+    @Override
+    public void flush() {
+        bytesStore.flush();
+    }
+
+    @Override
+    public void close() {
+        bytesStore.close();
+    }
+
+    @Override
+    public boolean persistent() {
+        return true;
+    }
+
+    @Override
+    public boolean isOpen() {
+        return bytesStore.isOpen();
+    }
+
+    @Override
+    public KeyValueIterator<Windowed<K>, AGG> fetch(final K key) {
+        return findSessionsToMerge(key, 0, Long.MAX_VALUE);
+    }
+
+    static class SessionKeySchema implements RocksDBSegmentedBytesStore.KeySchema {
+
+        @Override
+        public Bytes upperRange(final Bytes key, final long to) {
+            final Windowed<Bytes> sessionKey = new Windowed<>(key, new TimeWindow(to, Long.MAX_VALUE));
+            return SessionKeyBinaryConverter.toBinary(sessionKey, Serdes.Bytes().serializer());
+        }
+
+        @Override
+        public Bytes lowerRange(final Bytes key, final long from) {
+            final Windowed<Bytes> sessionKey = new Windowed<>(key, new TimeWindow(0, Math.max(0, from)));
+            return SessionKeyBinaryConverter.toBinary(sessionKey, Serdes.Bytes().serializer());
+        }
+
+        @Override
+        public long segmentTimestamp(final Bytes key) {
+            return SessionKeyBinaryConverter.extractEnd(key.get());
+        }
+
+        @Override
+        public HasNextCondition<Bytes> hasNextCondition(final Bytes binaryKey, final long from, final long to) {
+            return hasNextMatchingSession(from, to, binaryKey);
+        }
+
+        @Override
+        public List<Segment> segmentsToSearch(final Segments segments, final long from, final long to) {
+            return segments.segments(from, Long.MAX_VALUE);
+        }
+
+    }
+
+    static HasNextCondition<Bytes> hasNextMatchingSession(final long earliestEndTime, final long latestStartTime, final Bytes binaryKey) {
+        return new HasNextCondition<Bytes>() {
+            @Override
+            public boolean hasNext(final KeyValueIterator<Bytes, ?> iterator) {
+                if (iterator.hasNext()) {
+                    final Bytes bytes = iterator.peekNextKey();
+                    final Bytes keyBytes = Bytes.wrap(SessionKeyBinaryConverter.extractKeyBytes(bytes.get()));
+                    if (!keyBytes.equals(binaryKey)) {
+                        return false;
+                    }
+                    final long start = SessionKeyBinaryConverter.extractStart(bytes.get());
+                    final long end = SessionKeyBinaryConverter.extractEnd(bytes.get());
+                    return end >= earliestEndTime && start <= latestStartTime;
+                }
+                return false;
+            }
+        };
+    }
+
+    static class SessionStoreIterator<K, AGG> implements KeyValueIterator<Windowed<K>, AGG> {
```

Comment:
It doesn't strictly need to be part of `RocksDbSessionStore`, but it is only used here as of now. If it is needed elsewhere later, then we should move it later.